.github/workflows/build.yml

```diff
@@ -2,18 +2,40 @@ name: build
 
 on:
   workflow_call:
- 
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
 jobs:
   build:
     runs-on: ubuntu-latest
 
+    permissions:
+      contents: read
+      packages: none
+
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout repository
+        uses: actions/checkout@v3.5.3
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2.2.0
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Run a one-line script
-        run: echo Hello, world!
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v4.6.0
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
 
-      - name: Run a multi-line script
-        run: |
-          echo Add other actions to buil,
-          echo test, and deploy your project.
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v4.1.1
```

Comment:
no, we will use semantic release to publish images, like we do on containerbase.

Reply:
Agreed. Can we defer this change to a later stage?